pkgs/by-name/de/deepfilternet/package.nix

```diff
@@ -15,6 +15,11 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-5bYbfO1kmduNm9YV5niaaPvRIDRmPt4QOX7eKpK+sWY=";
   };
 
+  cargoPatches = [
+    # Fix compilation with Rust 1.80 (https://github.com/NixOS/nixpkgs/issues/332957)
+    ./cargo-lock-bump-time.patch
```

Comment:
Can we symlink the Cargo.lock into place, rather than having to have both a Cargo.lock file and a patch file?  Like in https://github.com/NixOS/nixpkgs/pull/334818.

Reply:
I kind of like this approach since its clearly bound to the package version that's being patched. It also communicates why the (updated) vendored Cargo.lock is necessary and what's changed.

It seems like the linked PR might cause strange situations when the underlying reason for the patch is made redundant by an upstream release, since there won't be a "patch doesn't apply anymore" to nudge maintainers.

Let me know if this seems reasonable to you. I'm new-ish to Nix packaging so my hunch might be off here :slightly_smiling_face:

Reply:
I think you're quite right, and this is a good instinct.  In this case, I think a comment would probably be enough to get it removed on update — it's right there below the source hash, which has to be updated anyway.  I regret not asking for that on the other PR.